### PR TITLE
Configure Branch Protection Settings via .github/settings.yml instead of UI

### PR DESCRIPTION
**Pull request by holdenhinkle**
References https://github.com/department-of-veterans-affairs/va.gov-team/issues/56437

This is a second attempt to configure this file.

Original PR - https://github.com/department-of-veterans-affairs/vets-api/pull/12631

***

Note from @rmtolmach regarding that PR:

vets-api PR was reverted. Support was getting alerted that people were not able to merge their PR's. These branch protections were meant to match what was currently being used, not lockdown merging. Revert PR: https://github.com/department-of-veterans-affairs/vets-api/pull/12659. But the settings didn't go back to what they were before we had that settings.yaml file, so we modified the settings in the UI to be what they used to be (we think!).

The two settings I unchecked in the UI were

“Restrict who can push to matching branches”
“Require branches to be up to date before merging”

***

Note from @holdenhinkle regarding that PR:

I'm guessing this section was causing the problems today:

```
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions:
        apps: []
        users: []
        teams: []
```

By setting `apps`, `users`, and `teams` to an empty array, we're telling GH that we want to restrict by `apps`, `users`, and `teams` but we're not telling GH who's allowed (I think allowed `apps`, `users`, and `teams` would be elements in the arrays). We should set each to `null` per the instructions:

> Set to null to disable.

```
      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
      restrictions:
        apps: null
        users: null
        teams: null
```

## What areas of the site does it impact?
Allowing engineers to merge their own code.

### CODE REVIEW

**Comment by rmtolmach** [comment]
Hey, Holden, I've converted this to a draft until it's ready to work on! We talked about Ryan J. picking it up.

**Review by rjohnson2011 — DISMISSED** [review]

File: .github/settings.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          - danger/danger
+      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
+      enforce_admins: true
+      # Prevent merge commits from being pushed to matching branches
+      required_linear_history: true
+      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
+      restrictions:
+        apps: null
+        users: null
+        teams: null
```
Comment (rmtolmach) [inline .github/settings.yml:38]: I'm comparing the settings in this file to the original PR: https://github.com/department-of-veterans-affairs/vets-api/pull/12631 and I see the `restrictions` are updated to `null`. 

When the original PR merged, it seemed like there were TWO settings that were wrong.
> The two settings I unchecked in the UI were:
> 1. “Restrict who can push to matching branches”
> 2. “Require branches to be up to date before merging”

Holden found that setting these values to `null` would fix number 1. @rjohnson2011 What will fix number 2? We always give the _option_ of updating a branch, but we don't want to make that _required_.

Comment (rjohnson2011) [reply .github/settings.yml]: @rmtolmach I updated the branch to make this `strict: false` (not requiring branches to be up to date before merging) as according to these docs -> https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch_protection

Thanks for catching that.

File: .github/settings.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          - danger/danger
+      # Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.
+      enforce_admins: true
+      # Prevent merge commits from being pushed to matching branches
+      required_linear_history: true
+      # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
+      restrictions:
+        apps: []
+        users: []
+        teams: []
```
Comment (rmtolmach) [inline .github/settings.yml:38]: wait... we _do_ want these to be `null` (according to Holden's research!). Srry if my last comment was confusing!

File: .github/settings.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        require_code_owner_reviews: true
+        # Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.
+        dismissal_restrictions:
+          users: []
+          teams: 
+            - platform-console-services
+      # Required. Require status checks to pass before merging. Set to null to disable
+      required_status_checks:
+        # Required. Require branches to be up to date before merging.
+        strict: false
```
Comment (rmtolmach) [inline .github/settings.yml:24]: nice! yeah, this looks right.

File: .github/settings.yml (hunk 2 above)
Comment (rjohnson2011) [reply .github/settings.yml]: Yes! Too much context switching, not sure how that happened. Thank you!!

**Pull request merged**